Test Ollama Rag Integration

```
Run cd src/praisonai && python -m pytest tests/integration/ -v --tb=short --disable-warnings
============================= test session starts ==============================
platform linux -- Python 3.11.13, pytest-8.4.1, pluggy-1.6.0 -- /opt/hostedtoolcache/Python/3.11.13/x64/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/PraisonAI/PraisonAI/src/praisonai
configfile: pytest.ini
plugins: langsmith-0.4.8, cov-6.2.1, asyncio-1.1.0, anyio-4.9.0
asyncio: mode=Mode.AUTO, asyncio_default_fixture_loop_scope=function, asyncio_default_test_loop_scope=function
collecting ... collected 38 items

tests/integration/autogen/test_autogen_basic.py::TestAutoGenIntegration::test_autogen_import PASSED [  2%]
tests/integration/autogen/test_autogen_basic.py::TestAutoGenIntegration::test_basic_autogen_agent_creation PASSED [  5%]
tests/integration/autogen/test_autogen_basic.py::TestAutoGenIntegration::test_autogen_conversation_flow PASSED [  7%]
tests/integration/autogen/test_autogen_basic.py::TestAutoGenIntegration::test_autogen_config_validation PASSED [ 10%]
tests/integration/crewai/test_crewai_basic.py::TestCrewAIIntegration::test_crewai_import PASSED [ 13%]
tests/integration/crewai/test_crewai_basic.py::TestCrewAIIntegration::test_basic_crewai_agent_creation PASSED [ 15%]
tests/integration/crewai/test_crewai_basic.py::TestCrewAIIntegration::test_crewai_crew_workflow PASSED [ 18%]
tests/integration/crewai/test_crewai_basic.py::TestCrewAIIntegration::test_crewai_config_validation PASSED [ 21%]
tests/integration/crewai/test_crewai_basic.py::TestCrewAIIntegration::test_crewai_agent_collaboration PASSED [ 23%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_llm_class_maps_base_url_to_api_base PASSED [ 26%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_agent_with_llm_dict_base_url_parameter PASSED [ 28%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_image_agent_base_url_consistency PASSED [ 31%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_koboldcpp_specific_scenario PASSED [ 34%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_litellm_documentation_example_compatibility PASSED [ 36%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_backward_compatibility_with_api_base PASSED [ 39%]
tests/integration/test_base_url_api_base_fix.py::TestBaseUrlApiBaseMapping::test_ollama_environment_variable_compatibility PASSED [ 42%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_server_connection PASSED [ 44%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_tool_execution PASSED [ 47%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_tool_wrapper PASSED [ 50%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_agent_with_mcp_tools PASSED [ 52%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_server_parameters PASSED [ 55%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_error_handling PASSED [ 57%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_multiple_servers PASSED [ 60%]
tests/integration/test_mcp_integration.py::TestMCPIntegration::test_mcp_tool_with_complex_parameters PASSED [ 63%]
tests/integration/test_mcp_integration.py::TestMCPAgentIntegration::test_agent_with_mcp_wrapper PASSED [ 65%]
tests/integration/test_mcp_integration.py::TestMCPAgentIntegration::test_mcp_async_tool_integration PASSED [ 68%]
tests/integration/test_mcp_integration.py::TestMCPAgentIntegration::test_mcp_tool_registry PASSED [ 71%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_rag_config_creation PASSED [ 73%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_agent_with_knowledge_config FAILED [ 76%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_vector_store_operations PASSED [ 78%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_knowledge_indexing_simulation PASSED [ 81%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_knowledge_retrieval_simulation PASSED [ 84%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_rag_agent_with_different_providers PASSED [ 86%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_ollama_rag_integration FAILED [ 89%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_rag_context_injection PASSED [ 92%]
tests/integration/test_rag_integration.py::TestRAGIntegration::test_multi_document_rag PASSED [ 94%]
tests/integration/test_rag_integration.py::TestRAGMemoryIntegration::test_rag_with_memory_persistence PASSED [ 97%]
tests/integration/test_rag_integration.py::TestRAGMemoryIntegration::test_rag_knowledge_update PASSED [100%]

=================================== FAILURES ===================================
_____________ TestRAGIntegration.test_agent_with_knowledge_config ______________
tests/integration/test_rag_integration.py:83: in test_agent_with_knowledge_config
    assert agent.knowledge is not None
E   assert None is not None
E    +  where None = <praisonaiagents.agent.agent.Agent object at 0x7f93f3ecfa10>.knowledge
----------------------------- Captured stdout call -----------------------------
           DEBUG    [11:29:18] llm.py:159 DEBUG LLM instance          llm.py:159
                    initialized with: {                                         
                      "model": "***",                                   
                      "timeout": null,                                          
                      "temperature": null,                                      
                      "top_p": null,                                            
                      "n": null,                                                
                      "max_tokens": null,                                       
                      "presence_penalty": null,                                 
                      "frequency_penalty": null,                                
                      "logit_bias": null,                                       
                      "response_format": null,                                  
                      "seed": null,                                             
                      "logprobs": null,                                         
                      "top_logprobs": null,                                     
                      "api_version": null,                                      
                      "stop_phrases": null,                                     
                      "api_key": "***",                                         
                      "base_url": null,                                         
                      "verbose": true,                                          
                      "markdown": true,                                         
                      "self_reflect": false,                                    
                      "max_reflect": 3,                                         
                      "min_reflect": 1,                                         
                      "reasoning_steps": false,                                 
                      "extra_settings": {                                       
                        "metrics": false                                        
                      }                                                         
                    }                                                           
------------------------------ Captured log call -------------------------------
DEBUG    root:llm.py:159 LLM instance initialized with: {
  "model": "***",
  "timeout": null,
  "temperature": null,
  "top_p": null,
  "n": null,
  "max_tokens": null,
  "presence_penalty": null,
  "frequency_penalty": null,
  "logit_bias": null,
  "response_format": null,
  "seed": null,
  "logprobs": null,
  "top_logprobs": null,
  "api_version": null,
  "stop_phrases": null,
  "api_key": "***",
  "base_url": null,
  "verbose": true,
  "markdown": true,
  "self_reflect": false,
  "max_reflect": 3,
  "min_reflect": 1,
  "reasoning_steps": false,
  "extra_settings": {
    "metrics": false
  }
}
________________ TestRAGIntegration.test_ollama_rag_integration ________________
tests/integration/test_rag_integration.py:259: in test_ollama_rag_integration
    assert agent.knowledge is not None
E   assert None is not None
E    +  where None = <praisonaiagents.agent.agent.Agent object at 0x7f93f33cbe90>.knowledge
=========================== short test summary info ============================
FAILED tests/integration/test_rag_integration.py::TestRAGIntegration::test_agent_with_knowledge_config - assert None is not None
 +  where None = <praisonaiagents.agent.agent.Agent object at 0x7f93f3ecfa10>.knowledge
FAILED tests/integration/test_rag_integration.py::TestRAGIntegration::test_ollama_rag_integration - assert None is not None
 +  where None = <praisonaiagents.agent.agent.Agent object at 0x7f93f33cbe90>.knowledge
================== 2 failed, 36 passed, 6 warnings in 12.44s ===================
Error: Process completed with exit code 1.
```

Identify the issue and find the fix. 

Is the issue with testing or issue with praisonaiagents application

--- a/src/praisonai/tests/integration/test_rag_integration.py
+++ b/src/praisonai/tests/integration/test_rag_integration.py
@@ -79,7 +79,11 @@ class TestRAGIntegration:
         
         assert agent.name == "RAG Knowledge Agent"
         assert hasattr(agent, 'knowledge')
-        # knowledge_config is passed to Knowledge constructor, not stored as attribute
+        # With lazy loading, knowledge is None until first use
+        assert agent._knowledge_sources is not None
+        assert agent._knowledge_processed == False
+        # Trigger knowledge processing
+        agent._ensure_knowledge_processed()
         assert agent.knowledge is not None
     
     @patch('chromadb.Client')
@@ -256,6 +260,11 @@ class TestRAGIntegration:
         
         assert agent.name == "Ollama RAG Agent"
         assert hasattr(agent, 'knowledge')
+        # With lazy loading, knowledge is None until first use
+        assert agent._knowledge_sources is not None
+        assert agent._knowledge_processed == False
+        # Trigger knowledge processing
+        agent._ensure_knowledge_processed()
         assert agent.knowledge is not None
     
     @patch('chromadb.Client')

